main.cpp

```diff
@@ -133,256 +175,335 @@ struct CarWash
-    // 2) spin independently of wheel
-    void freeSpin();
-    // 3) shift gear ratio
-    int shiftGear(); // returns the current gear
+    bool needMaintenance = false;
+
+    int propelBike(bool pedal, int gearRatio, int pedalSpeed);
+    bool freeSpin(bool needMaintenance);
+    int shiftGear(int leftShifterValue, int rightShifterValue); 
 };
 
+int Gears::propelBike(bool pedal, int gearRatio, int pedalSpeed)
+{
+    return pedal ? gearRatio * pedalSpeed : 0;
+}
+
+bool Gears::freeSpin(bool needmaintenanceGears)
+{
+    return needmaintenanceGears ? false : true;
+}
+
+int Gears::shiftGear(int leftShifterValue, int rightShifterValue)
+{
+    return leftShifterValue * rightShifterValue;    
+}
+
 struct Frame
 {
-    // weight
     float weight = 3.7f;
-    // type
     char type = 'A';
-    // material
-    std::string material = "Carbon Fiber";
-    // brand    
+    std::string material = "Carbon Fiber"; 
     std::string brand = "Acme";
-    // need maintenance?
-    bool needmaintenance = false;
-
-    // 3 things it can do:
-    // 1) adjust seat height
-    int adjustSeatHight(int riderHeight); // returns the height of the seat
-    // 2) absorb shock from bumps
-    void shockAbsorb();
-    // 3) support weight of rider
-    void supportWeight();
+    bool needMaintenance = false;
+
+    int adjustSeatHeight(int riderHeight);
+    bool shockAbsorbFrame(bool needMaintenance);
+    bool supportWeight(bool needMaintenance);
 };
 
+int Frame::adjustSeatHeight(int riderHeight)
+{
+    return ((riderHeight / 70) + 2 ) * 3;
+}
+
+bool Frame::shockAbsorbFrame(bool needMaintenanceFrame)
+{
+    return needMaintenanceFrame ? false : true;
+}
+
+bool Frame::supportWeight(bool needMaintenanceFrame)
+{
+    return needMaintenanceFrame ? false : true;
+}
+
 struct Wheel
 {
-    // circumference
     float circumference = 30.2f;
-    // number of spokes
     int numberOfSpokes = 25;
-    // material
     std::string material = "Aluminum";
-    // brand
     std::string brand = "Acme";
-    // need maintenance?
-    bool needmaintenance = false;
-
-    // 3 things it can do:
-    // 1) spin
-    int spin(int speed); // returns spin duration 
-    // 2) absorb shock from bumps
-    void shockAbsorb();
-    // 3) support weight of rider
-    void supportWeight();
+    bool needMaintenance = false;
+
+    int spin(int speed, int distance); 
+    bool shockAbsorbWheel(bool needMaintenance);
+    bool supportWeightWheel(bool needMaintenance);
 };
 
+int Wheel::spin(int speed, int distance)
+{
+    return speed * distance;
+}
+
+bool Wheel::shockAbsorbWheel(bool needMaintenanceWheel)
+{
+    return needMaintenanceWheel ? false : true;
+}
+
+bool Wheel::supportWeightWheel(bool needMaintenanceWheel)
+{
+    // should I write it like this or how it was above? I feel like this way is less confusing with the true value being first spot in the ternary opeator after the question mark.
+    return needMaintenanceWheel == false ? true : false;
+}
```

Comment:
For these, you can just write:
```
return ! needMaintenanceWheel;
```
This returns the inverse boolean value that `needMaintenanceWheel` has.

- [ ] revise

Reply:
Nice